platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/SolrCloudClientFactory.java

```diff
@@ -0,0 +1,209 @@
+        } catch (SolrFactoryException e) {
+            LOGGER.debug("Unable to upload configuration to Solr Cloud", e);
+            return null;
+        }
+
+        try {
+            createCollection(collection, client);
+        } catch (SolrFactoryException e) {
+            LOGGER.debug("Unable to create collection on Solr Cloud", e);
+            return null;
+        }
+
+        client.setDefaultCollection(collection);
+        return client;
+    }
+
+    private static void createCollection(String collection, CloudSolrClient client)
+            throws SolrFactoryException {
+        try {
+            CollectionAdminResponse response = new CollectionAdminRequest.List().process(client);
+
+            if (response == null || response.getResponse() == null || response.getResponse()
+                    .get("collections") == null) {
+                throw new SolrFactoryException("Failed to get a list of existing collections");
+            }
+
+            List<String> collections = (List<String>) response.getResponse()
+                    .get("collections");
+            if (!collections.contains(collection)) {
+                response = new CollectionAdminRequest.Create().setNumShards(SHARD_COUNT)
+                        .setMaxShardsPerNode(MAXIMUM_SHARDS_PER_NODE)
+                        .setReplicationFactor(REPLICATION_FACTOR)
+                        .setCollectionName(collection)
+                        .process(client);
+                if (!response.isSuccess()) {
+                    throw new SolrFactoryException(
+                            "Failed to create collection [" + collection + "]: "
+                                    + response.getErrorMessages());
+                }
+                isCollectionReady(client, collection);
+            } else {
+                LOGGER.debug("Collection already exists: " + collection);
+            }
+        } catch (SolrServerException | IOException e) {
+            throw new SolrFactoryException("Failed to create collection: " + collection, e);
+        }
+    }
+
+    private static void uploadCoreConfiguration(String collection, CloudSolrClient client)
+            throws SolrFactoryException {
+        boolean configExistsInZk;
+
+        try {
+            configExistsInZk = client.getZkStateReader()
+                    .getZkClient()
+                    .exists("/configs/" + collection, true);
+        } catch (KeeperException | InterruptedException e) {
+            throw new SolrFactoryException("Failed to check config status with Zookeeper for collection: " + collection,
+                    e);
+        }
+
+        if (!configExistsInZk) {
+            ConfigurationFileProxy configProxy =
+                    new ConfigurationFileProxy(ConfigurationStore.getInstance());
+            configProxy.writeSolrConfiguration(collection);
+            Path configPath = Paths.get(configProxy.getDataDirectory()
```

Comment:
Would it make sense to add a `Path getConfigPath`method on the `ConfigurationFileProxy` class instead of having this code here?